### PR TITLE
Report conflicts when renaming a partial method definition's parameters

### LINKED ISSUE #18566: Conflict is not indicated when renaming a parameter from partial method definition
**Version Used**: Visual Studio 2017 15.0.0+26228.10

**Code to Reproduce**:
```csharp
public partial class Foo
{
    partial void Bar(Foo value)
    {
        Foo foo = null;
    }
}

public partial class Foo
{
    partial void Bar(Foo value); // Rename 'value' to 'foo' from here
}
```
**Actual Behavior**:
Conflict is not indicated in the IDE.`

### PR DESCRIPTION
**Pull request by tuespetre**
Fixes #18566

### CODE REVIEW

File: src/Workspaces/CSharp/Portable/Rename/CSharpRenameRewriterLanguageService.cs [hunk 1]
```diff
@@ -814,19 +814,34 @@ private SyntaxToken RenameWithinToken(SyntaxToken oldToken, SyntaxToken newToken
...
-                    visitor.Visit(methodDeclaration);
+
+                    visitor.Visit(memberDeclaration);
                     conflicts.AddRange(visitor.ConflictingTokens.Select(t => reverseMappedLocations[t.GetLocation()]));
+
+                    // If this is a parameter symbol for a partial method definition, be sure we visited 
+                    // the implementation part's body.
+                    if (renamedSymbol is IParameterSymbol renamedParameterSymbol &&
+                        renamedSymbol.ContainingSymbol is IMethodSymbol methodSymbol &&
+                        methodSymbol.PartialImplementationPart is IMethodSymbol partialImplementationPart)
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/Rename/CSharpRenameRewriterLanguageService.cs]: PartialImplementationPart is always an IMethodSymbol.  unless you're doing this for a null check, in which case, i would just do a null check.

Comment (tuespetre) [inline src/Workspaces/CSharp/Portable/Rename/CSharpRenameRewriterLanguageService.cs]: Sure thing, seemed natural earlier but coming back to it, a simple null check would be clearer :+1:

**Comment by Pilchie** [comment]
retest windows_release_vs-integration_prtest please

**Comment by jcouv** [comment]
@Pilchie @dpoeschl What's holding back from merging this contribution?
(Note, I updated milestone from 15.5 to 15.7)

**Comment by dpoeschl** [comment]
Thanks for bringing this back up @jcouv. This looks good to me.

**Comment by jcouv** [comment]
@dpoeschl No problem. Will you be merging? Is `master` the right branch?

**Comment by dpoeschl** [comment]
@jinujoseph This is a fix that we'd like to get in when convenient (unless @jcouv has a more pressing need for it), but we don't need to rush it into a particular release. Julien was thinking 15.7. Does that sound good? Do we need any further approval?

**Comment by jinujoseph** [comment]
@Pilchie from ask mode perspective

**Comment by jcouv** [comment]
Since the PR is targeting `master`, it's probably more natural to just leave it there and update milestone to 15.8.

**Comment by Pilchie** [comment]
Either is fine with me.

**Comment by jcouv** [comment]
CI seems to be stuck (no status reported for the coreclr legs). I'll close and re-open the PR to kick it.

**Comment by jinujoseph** [comment]
@dotnet-bot retest this please

**Comment by dpoeschl** [comment]
This looks good to merge to me, as soon as tests are passing. 👍

**Comment by dpoeschl** [comment]
@jinujoseph I'm not sure what's happening with this over the last few days. Anything I can do to help?

**Comment by tuespetre** [comment]
re🅱️ased for great justice

**Comment by jinujoseph** [comment]
finally :-)

**Comment by jinujoseph** [comment]
approved to merge for 15.8.preview4

**Comment by dpoeschl** [comment]
Thanks @tuespetre!!!

**Pull request merged**